### PR TITLE
feat(packages/backend-common): allow to provide authentification to DockerContainerRunner

### PR DESCRIPTION
**Pull request by secustor**
## Hey, I just made a Pull Request!
Introduce a new field in `DockerRunOptions` which allows to provide the credentials. 

I have considered setting up a new facility to provide OCI credentials to the system, tough this would more discussion or even a BEP. 
This is a stop gap solution, tough can be integrated in a later one. E.g. credentials are provided by integrations by default, but users can overwrite authentication via DockerRunOptions. 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by vinzscam — CHANGES_REQUESTED** [review]

File: .changeset/tame-pianos-hunt.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/backend-common': minor
+---
+
+Allow providing authentication which in turn allows usage of private registries
```
Comment (vinzscam) [inline .changeset/tame-pianos-hunt.md:5]:
```suggestion
Added `pullOptions` to `DockerContainerRunner#runContainer` method to pass down options when pulling an image.
```

**Review by vinzscam — CHANGES_REQUESTED** [review]
nice, tweaked the changeset a bit then it's ready to go 😄

**Review by vinzscam — APPROVED** [review]

File: .changeset/tame-pianos-hunt.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/backend-common': minor
```
Comment (Rugvip) [inline .changeset/tame-pianos-hunt.md:2]:
```suggestion
'@backstage/backend-common': patch
```

No breaking changes so bringing it down to `patch` for a `0.x` package

**Review by Rugvip — APPROVED** [review]
Thank you! 👍

**Pull request merged**